### DISCUSSION TITLE
For the "Create and assign property" quick-action, how can I change the default accessibility for generated properties?

### DISCUSSION THREAD

**Discussion by Metritutus** [dotnet/roslyn#71422] [Q&A] [answered]
A lot of the time, I'm using the "Create and assign property" quick-action to generate a property for a service I'm injecting via the constructor.  I don't want it to be public, I want it to be private.  While I can obviously manually update the accessibility after the code is generated, I'd rather have it default to private.  However, I simply can't figure out how to change the default accessibility of the generated property to be private.  Is there any way to customise this behaviour?

**Comment by CyrusNajmabadi** [comment, marked as answer]
There is no way to make this private.  The expectation is that if you want a property, it will be public (That's the 99% case).  If you want private, ask to make a field, which will then be private (again, the 99% case).

**Reply by Metritutus** [reply]
Ah darn, I'd hoped there was some customisation available via snippets or .editorconfig, etc?

**Reply by Metritutus** [reply]
For future readers in the aforementioned 1% case, ~~I _believe_ the code that specifies the accessibility is in the `CreateProperty()` method in `CSharpInitializeMemberFromPrimaryConstructorParameterCodeRefactoringProvider.cs`, [here](https://github.com/dotnet/roslyn/blob/e18df20f86680d58ff0f7d563969f8ff28f8640f/src/Features/CSharp/Portable/InitializeParameter/CSharpInitializeMemberFromPrimaryConstructorParameterCodeRefactoringProvider.cs#L241), where the accessibility is hard-coded, so as the official reply above states, there's no way to change the default accessibility of the generated property.~~

EDIT:

After further investigation, my above message is incorrect on the location of the code in question.  The file mentioned looks like it might be a work-in-progress.  The actual functionality used by Visual Studio right now appears to be the `CreatePropertySymbol()` method in `CSharpGenerateTypeService.cs`, [here](https://github.com/dotnet/roslyn/blob/9d80dea7fe1b14043b9b2ac4d0b59ed26f508742/src/Features/CSharp/Portable/GenerateType/CSharpGenerateTypeService.cs#L898), however as with the previous file, the default accessibility is still hard-coded.
